File: fathersDay/Dsa/Day 22 (58. Length of Last Word)/Solution.java
class Solution {
    public int LengthOfLastWord(String s) {
        s = s.trim(); // Remove leading/trailing spaces
        if (s.isEmpty()) {
            return 0;
        }
        int lastSpaceIndex = s.lastIndexOf(' ');
        return s.length() - 1 - lastSpaceIndex;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        String s = "Hello World";
        System.out.println(solution.LengthOfLastWord(s)); // Output: 5
    }
}

File: fathersDay/Dsa/Day 22 (58. Length of Last Word)/Solution.cpp
#include <iostream>
#include <string>
#include <algorithm>

class Solution {
public:
    int LengthOfLastWord(std::string s) {
        // Trim trailing spaces
        int end = s.length() - 1;
        while (end >= 0 && s[end] == ' ') {
            end--;
        }
        if (end < 0) {
            return 0;
        }
        // Find the start of the last word
        int start = end;
        while (start >= 0 && s[start] != ' ') {
            start--;
        }
        return end - start;
    }
};

int main() {
    Solution solution;
    std::string s = "Hello World";
    std::cout << solution.LengthOfLastWord(s) << std::endl; // Output: 5
    return 0;
}
